lang/src/OTS/ITF/org/partiql/ots/type/ScalarType.kt

```diff
@@ -9,10 +8,40 @@ import org.partiql.lang.eval.ExprValueType
  */
 interface ScalarType {
     /**
-     * a type ID
+     * The type id, recognized by users or developers to know which type it is. One example use case
+     * is that, if we want to throw an error mentioning a scalar type in the error message, we can
+     * refer to its [id].
+     *
+     * Types cannot share the same type id
      */
     val id: String
 
+    /**
+     * Type names, recognized by program (PartiQL compiler). One use case is that, e.g., in the expression
+     * `CAST(1 AS <type_names>)`, the target type of the CAST operator corresponds to the scalar type whose
+     * [names] has `<type_name>`
+     *
+     * Note that spaces are not allowed in type aliases. For special type aliases with space in standard sql, e.g.
+     * `TIME WITH TIME ZONE` in the query `CAST(a AS TIME WITH TIME ZONE)`, the spaces there are replaced with
+     * underscore '_' during parsing, so the query is actually transformed as `CAST(a AS TIME_WITH_TIME_ZONE)`.
+     * All the special type aliases are:
+     *      1. CHARACTER VARYING -> CHARACTER_VARYING
+     *      2. DOUBLE PRECISION -> DOUBLE_PRECISION
+     *      3. TIME WITH TIME ZONE -> TIME_WITH_TIME_ZONE
```

Comment:
Because currently, PartiQL syntax does not allow spaces in type name, except for special type names, such as `TIME WITH TIME ZONE`.

Reply:
like `CAST(1 AS MY TYPE)` throws a syntax on `MY TYPE`

Reply:
Ok, so the validation should assert that names do not included spaces and our parser should be able to handle the special cases. By introducing the rewrite of " " to "_" then we can't reliably biject since there is no distinction between an explicity "_" or a lexer/parser inserted "_"

Reply:
I agree with you that "_" insertion is unnecessary.